qiskit/quantum_info/operators/symplectic/base_pauli.py

```diff
@@ -0,0 +1,639 @@
+
+        Returns:
+            BasePauli: the output Pauli.
+
+        Raises:
+            QiskitError: if number of qubits of other does not match qargs.
+        """
+        return self.compose(other, qargs=qargs, front=True, inplace=inplace)
+
+    def _multiply(self, other):
+        """Multiply each Pauli in the table by a phase.
+
+        Args:
+            other (complex): a complex number in [1, -1j, -1, 1j]
+
+        Returns:
+            BasePauli: the Pauli table other * self.
+
+        Raises:
+            QiskitError: if the phase is not in the set [1, -1j, -1, 1j].
+        """
+        if isinstance(other, (np.ndarray, list, tuple)):
+            phase = np.array([self._phase_from_complex(phase) for phase in other])
+        else:
+            phase = self._phase_from_complex(other)
+        return BasePauli(self._z, self._x, np.mod(self._phase + phase, 4))
+
+    def conjugate(self):
+        """Return the conjugate of each Pauli in the list."""
+        complex_phase = np.mod(self._phase, 2)
+        if np.all(complex_phase == 0):
+            return self
+        return BasePauli(self._z, self._x, np.mod(self._phase + 2 * complex_phase, 4))
+
+    def transpose(self):
+        """Return the transpose of each Pauli in the list."""
+        # Transpose sets Y -> -Y. This has effect on changing the phase
+        parity_y = self._count_y() % 2
+        if np.all(parity_y == 0):
+            return self
+        return BasePauli(self._z, self._x, np.mod(self._phase + 2 * parity_y, 4))
+
+    def commutes(self, other, qargs=None):
+        """Return True if Pauli that commutes with other.
+
+        Args:
+            other (BasePauli): another BasePauli operator.
+            qargs (list): qubits to apply dot product on (default: None).
+
+        Returns:
+            np.array: Boolean array of True if Pauli's commute, False if
+                      they anti-commute.
+
+        Raises:
+            QiskitError: if number of qubits of other does not match qargs.
+        """
+        if qargs is not None and len(qargs) != other.num_qubits:
+            raise QiskitError(
+                "Number of qubits of other Pauli does not match number of "
+                "qargs ({} != {}).".format(other.num_qubits, len(qargs)))
+        if qargs is None and self.num_qubits != other.num_qubits:
+            raise QiskitError(
+                "Number of qubits of other Pauli does not match the current "
+                "Pauli ({} != {}).".format(other.num_qubits, self.num_qubits))
+        if qargs is not None:
+            inds = list(qargs)
+            x1, z1 = self._x[:, inds], self._z[:, inds]
+        else:
+            x1, z1 = self._x, self._z
+        a_dot_b = np.mod(np.sum(np.logical_and(x1, other._z), axis=1), 2)
+        b_dot_a = np.mod(np.sum(np.logical_and(z1, other._x), axis=1), 2)
+        return a_dot_b == b_dot_a
+
+    def evolve(self, other, qargs=None):
```

Comment:
I prefer time evolution by Heisenberg picture here because Paulis are usually used for quantum observables. So, how about C^\dagger .P.C ? This is just my preference, but would you please consider it once.

Reply:
I'm fine with doing Heisenberg convention for operators